tests/Form/ChoiceList/ModelChoiceLoaderTest.php

```diff
@@ -17,14 +17,30 @@
 use Sonata\AdminBundle\Form\ChoiceList\ModelChoiceLoader;
 use Sonata\AdminBundle\Model\ModelManagerInterface;
 use Sonata\AdminBundle\Tests\Fixtures\Bundle\Entity\Foo;
+use Symfony\Bridge\PhpUnit\ExpectDeprecationTrait;
 
 class ModelChoiceLoaderTest extends TestCase
 {
+    use ExpectDeprecationTrait;
+
     private $modelManager;
 
     protected function setUp(): void
     {
-        $this->modelManager = $this->getMockForAbstractClass(ModelManagerInterface::class);
+        $this->modelManager = $this->createMock(ModelManagerInterface::class);
+    }
+
+    /**
+     * NEXT_MAJOR: Check for a InvalidArgumentException instead of the deprecation.
+     *
+     * @group legacy
+     */
+    public function testConstructWithUnsupportedQuery(): void
+    {
+        $this->modelManager->method('supportQuery')->willReturn(false);
+
+        $this->expectDeprecation('Passing a query which is not supported by the modelManager is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in version 4.0.');
```

Comment:
```suggestion
        $this->expectDeprecation('Passing a query which is not supported by the model manager is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in version 4.0.');
```

Reply:
Done